From 32135a4af7c2ea1db2ea528df9af0a331ca88dba Mon Sep 17 00:00:00 2001
From: "zan@falconsigh.net"
 <zan@falconsigh.net@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Tue, 12 Apr 2022 06:34:44 +0000
Subject: [PATCH] [GTK][WPE] Build ANGLE with the
 EGL_NO_PLATFORM_SPECIFIC_TYPES define
 https://bugs.webkit.org/show_bug.cgi?id=239039

Reviewed by Adrian Perez de Castro.

Specify the EGL_NO_PLATFORM_SPECIFIC_TYPES define when building ANGLE
subproject for the GTK and WPE ports. This should avoid searching for
platform-specific headers that might not be available at all during
build, e.g. the X11 headers which are used by default on UNIX platforms.

* PlatformGTK.cmake:
* PlatformWPE.cmake:

Signed-off-by: Adrian Perez de Castro <aperez@igalia.com>
Upstream status: https://bugs.webkit.org/show_bug.cgi?id=239039
---
 Source/ThirdParty/ANGLE/ChangeLog         | 15 +++++++++++++++
 Source/ThirdParty/ANGLE/PlatformGTK.cmake |  2 +-
 Source/ThirdParty/ANGLE/PlatformWPE.cmake |  2 +-
 3 files changed, 17 insertions(+), 2 deletions(-)

diff --git a/Source/ThirdParty/ANGLE/PlatformGTK.cmake b/Source/ThirdParty/ANGLE/PlatformGTK.cmake
index 6d5287f1900..edf72f99dcd 100644
--- a/Source/ThirdParty/ANGLE/PlatformGTK.cmake
+++ b/Source/ThirdParty/ANGLE/PlatformGTK.cmake
@@ -1,4 +1,4 @@
-list(APPEND ANGLE_DEFINITIONS ANGLE_PLATFORM_LINUX USE_SYSTEM_EGL)
+list(APPEND ANGLE_DEFINITIONS ANGLE_PLATFORM_LINUX EGL_NO_PLATFORM_SPECIFIC_TYPES USE_SYSTEM_EGL)
 include(linux.cmake)
 
 if (USE_OPENGL)
-- 
2.36.1

